OpenTrons/opentrons-api
# feature(api): Allow to move_labware off deck

# Overview

closes https://opentrons.atlassian.net/browse/RLAB-165.
Allow `move_labware` off-deck via papi.

# Test Plan

- Upload the following protocol via the app on an OT-3:
```
from typing import cast
from opentrons import protocol_api, types

metadata = {
    "apiLevel": "2.15",
}

def run(ctx: protocol_api.ProtocolContext) -> None:
    labware = ctx.load_labware("nest_96_wellplate_100ul_pcr_full_skirt", 2)
    pipette = ctx.load_instrument("p10_single", types.Mount.LEFT)
    ctx.move_labware(labware=labware, new_location=types.OFF_DECK)
```
- Verified that analysis passes
- Run the protocol. make sure there is a pause to extract the labware.


# Changelog

- Added an enum for `OffDeckType`
- Add `OffDeckType` to `new_location` Union in `move_labware`. gated behind 2.15 api version. 
- wire up core/engine to use `offDeck` literal for the PE `moveLabware` command.


# Review requests

Does these changes make sense? Did I miss anything?

# Risk assessment

low. added an arg type to `move_labware` in PAPI.
